### ISSUE TITLE
Disable the auto-generation of section IDs ?

### ISSUE THREAD

**Issue by lcawl** [elastic/docs#668]
While reviewing the differences between the old and new (Asciidoctor) output for https://github.com/elastic/docs/pull/666, we noticed that there are a *lot* of autogenerated section IDs.

Per https://asciidoctor.org/docs/user-manual/#auto-generated-ids, you can disable the auto-generation of section IDs, by unsetting the sectids attribute.

Should we consider doing this? It would mean that if we want to be able to link to a section, we'd have to explicitly add an anchor, which is what we had discussed as a best practice anyway.

If we do want to do this, would then want to consider whether we do it for all books or a subset.

Related to https://github.com/elastic/docs/issues/647

**Comment by nik9000** [comment]
I had a thought on this topic this morning. I wonder if we should keep generating the ID but detect when someone is linking to them and float some text on the screen explaining that the ID isn't permanent but that they can open a PR to make a permanent ID. Since most of our docs are public this is a thing that folks outside of Elastic the company can do. That way we can *know* that any non-auto-generated IDs are there for some reason. We can even `git blame` to figure out who made them.

**Comment by lcawl** [comment]
If that's feasible, it sounds pretty magical

**Comment by nik9000** [comment]
> If that's feasible, it sounds pretty magical

We'd have to use a heuristic to detect if an ID was auto-generated or not. So long as the heuristic is good I think we can do it. And I *think* "does it start with `_` is a good heuristic. We could also fail the docs build if you declare any IDs that start with `_` and tell you that those look too much like auto-generated-ids. At least, we can do that in Asciidoctor.

**Comment by nik9000** [comment]
@dedemorton mentioned today that page names also come from the IDs. I hadn't really thought of that for https://github.com/elastic/docs/issues/668#issuecomment-470147536 . My point in that issue was about the `id`s on the page, not the ids that turn into the page's name. I think we 100% should make folks specify the name of every page.

Another point from @nrichers an @kellyemurphy: some of our docs won't have an edit link because they are private. They'll need different language about how to ask to make the link real. Assuming we do want to generate these at all.

**Comment by debadair** [comment]
We absolutely should avoid generating pages with automatically generated IDs. Magic would be nice, but disabling them would at least keep from perpetuating the problem.

**Comment by nik9000** [comment]
The real reason I haven't picked this up is because I haven't had time yet.
I think soon we should do some prioritizing. I see the light at the end of
both of my current projects so it'll soon be time to start in on these.

**Comment by dedemorton** [comment]
> We absolutely should avoid generating pages with automatically generated IDs.

Agreed. I suspect that most of these were fixed during the migration to asciidoctor. I think we should allow auto-generated section IDs, though, unless Nik can create a script that we can run against the docs...or hire an intern. I never want to have to backport to 25 branches ever again.

**Comment by nik9000** [comment]
I think it'd be a "from here on out" thing.
